src/ngraph/runtime/backend_manager.hpp

```diff
@@ -0,0 +1,61 @@
+/*******************************************************************************
+* Copyright 2017-2018 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include <functional>
+#include <map>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#ifdef WIN32
+#include <windows.h>
+#define DL_HANDLE HMODULE
+#else
+#define DL_HANDLE void*
+#endif
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        class Backend;
+        class BackendManager;
+
+        using new_backend_t = std::function<Backend*(const char* config)>;
+    }
+}
+
+class ngraph::runtime::BackendManager
```

Comment:
Need to provide comments explaining how to use backend, how to specify one etc.

Reply:
I added comments for all of the public API